Thanks @legoktm for your feedback! I'll work on this tomorrow. 

If it's OK for me to ask, I have a couple of questions: 

- Going forward, in which scenarios should I group several changes into a single commit (and so removing the need of "squashing them down" once a PR is submitted)?
- I need to learn more about mypy, but the output of the CI linter is a bit puzzling (mostly because it picks on three statements and overlooks many other -identical- instances). Where should I start looking to solve this?
```
securedrop/i18n_tool.py:275: error: No overload variant of "run" matches argument types "List[str]", "Dict[str, str]"
securedrop/i18n_tool.py:275: note: Possible overload variants:
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[str] = ..., text: Literal[True], timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: str, errors: Optional[str] = ..., input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: str, input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., *, universal_newlines: Literal[True], startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: Literal[False] = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: None = ..., errors: None = ..., input: Optional[bytes] = ..., text: Optional[Literal[False]] = ..., timeout: Optional[float] = ...) -> CompletedProcess[bytes]
securedrop/i18n_tool.py:275: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[Union[bytes, str]] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[Any]
securedrop/i18n_tool.py:277: error: No overload variant of "run" matches argument types "List[str]", "Dict[str, str]"
securedrop/i18n_tool.py:277: note: Possible overload variants:
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[str] = ..., text: Literal[True], timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: str, errors: Optional[str] = ..., input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: str, input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., *, universal_newlines: Literal[True], startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[str] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[str]
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: Literal[False] = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: None = ..., errors: None = ..., input: Optional[bytes] = ..., text: Optional[Literal[False]] = ..., timeout: Optional[float] = ...) -> CompletedProcess[bytes]
securedrop/i18n_tool.py:277: note:     def run(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stdout: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, capture_output: bool = ..., check: bool = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., input: Optional[Union[bytes, str]] = ..., text: Optional[bool] = ..., timeout: Optional[float] = ...) -> CompletedProcess[Any]
securedrop/i18n_tool.py:279: error: No overload variant of "check_output" matches argument types "List[str]", "str", "Dict[str, str]"
securedrop/i18n_tool.py:279: note: Possible overload variants:
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., text: Literal[True]) -> str
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: str, errors: Optional[str] = ..., text: Optional[bool] = ...) -> str
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: Optional[str] = ..., errors: str, text: Optional[bool] = ...) -> str
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., *, universal_newlines: Literal[True], startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., text: Optional[bool] = ...) -> str
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: Literal[False] = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: None = ..., errors: None = ..., text: Optional[Literal[False]] = ...) -> bytes
securedrop/i18n_tool.py:279: note:     def check_output(args: Union[Union[str, bytes, PathLike[str], PathLike[bytes]], Sequence[Union[str, bytes, PathLike[str], PathLike[bytes]]]], bufsize: int = ..., executable: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., stdin: Union[None, int, IO[Any]] = ..., stderr: Union[None, int, IO[Any]] = ..., preexec_fn: Optional[Callable[[], Any]] = ..., close_fds: bool = ..., shell: bool = ..., cwd: Optional[Union[str, bytes, PathLike[str], PathLike[bytes]]] = ..., env: Optional[Union[Mapping[bytes, Union[str, bytes, PathLike[str], PathLike[bytes]]], Mapping[str, Union[str, bytes, PathLike[str], PathLike[bytes]]]]] = ..., universal_newlines: bool = ..., startupinfo: Any = ..., creationflags: int = ..., restore_signals: bool = ..., start_new_session: bool = ..., pass_fds: Any = ..., *, timeout: Optional[float] = ..., input: Optional[Union[bytes, str]] = ..., encoding: Optional[str] = ..., errors: Optional[str] = ..., text: Optional[bool] = ...) -> Any
```

--- a/Makefile
+++ b/Makefile
@@ -108,8 +108,6 @@ app-lint:  ## Test pylint compliance.
 	@echo "███ Linting application code..."
 	@cd securedrop && find . -name '*.py' | xargs pylint --reports=no --errors-only \
 	   --disable=no-name-in-module \
-	   --disable=unexpected-keyword-arg \
-	   --disable=too-many-function-args \
 	   --disable=import-error \
 	   --disable=no-member \
 	   --max-line-length=100

--- a/securedrop/i18n_tool.py
+++ b/securedrop/i18n_tool.py
@@ -18,7 +18,6 @@ from pathlib import Path
 from typing import Dict, List, Optional, Set
 
 import version
-from sh import git, msgfmt, msgmerge, pybabel, xgettext
 
 logging.basicConfig(format="%(asctime)s %(levelname)s %(message)s")
 log = logging.getLogger(__name__)
@@ -59,10 +58,10 @@ class I18NTool:
         Make sure we have a git remote for the i18n repo.
         """
 
-        k = {"_cwd": args.root}
-        if b"i18n" not in git.remote(**k).stdout:
-            git.remote.add("i18n", args.url, **k)
-        git.fetch("i18n", **k)
+        k = {"cwd": args.root}
+        if "i18n" not in subprocess.check_output(["git", "remote"], encoding="utf-8", **k):
+            subprocess.check_call(["git", "remote", "add", "i18n", args.url], **k)
+        subprocess.check_call(["git", "fetch", "i18n"], **k)
 
     def translate_messages(self, args: argparse.Namespace) -> None:
         messages_file = Path(args.translations_dir).absolute() / "messages.pot"
@@ -71,22 +70,26 @@ class I18NTool:
             if not os.path.exists(args.translations_dir):
                 os.makedirs(args.translations_dir)
             sources = args.sources.split(",")
-            pybabel.extract(
-                "--charset=utf-8",
-                "--mapping",
-                args.mapping,
-                "--output",
-                str(messages_file),
-                "--project=SecureDrop",
-                "--version",
-                args.version,
-                "--msgid-bugs-address=securedrop@freedom.press",
-                "--copyright-holder=Freedom of the Press Foundation",
-                "--add-comments=Translators:",
-                "--strip-comments",
-                "--add-location=never",
-                "--no-wrap",
-                *sources,
+            subprocess.check_call(
+                [
+                    "pybabel",
+                    "extract",
+                    "--charset=utf-8",
+                    "--mapping",
+                    args.mapping,
+                    "--output",
+                    messages_file,
+                    "--project=SecureDrop",
+                    "--version",
+                    args.version,
+                    "--msgid-bugs-address=securedrop@freedom.press",
+                    "--copyright-holder=Freedom of the Press Foundation",
+                    "--add-comments=Translators:",
+                    "--strip-comments",
+                    "--add-location=never",
+                    "--no-wrap",
+                    *sources,
+                ]
             )
 
             msg_file_content = messages_file.read_text()
@@ -101,31 +104,42 @@ class I18NTool:
             ):
                 tglob = "{}/*/LC_MESSAGES/*.po".format(args.translations_dir)
                 for translation in glob.iglob(tglob):
-                    msgmerge("--previous", "--update", "--no-wrap", translation, messages_file)
+                    subprocess.check_call(
+                        [
+                            "msgmerge",
+                            "--previous",
+                            "--update",
+                            "--no-wrap",
+                            translation,
+                            messages_file,
+                        ]
+                    )
                 log.warning(f"messages translations updated in {messages_file}")
             else:
                 log.warning("messages translations are already up to date")
 
         if args.compile and len(os.listdir(args.translations_dir)) > 1:
-            pybabel.compile("--directory", args.translations_dir)
+            subprocess.check_call(["pybabel", "compile", "--directory", args.translations_dir])
 
     def translate_desktop(self, args: argparse.Namespace) -> None:
         messages_file = Path(args.translations_dir).absolute() / "desktop.pot"
 
         if args.extract_update:
             sources = args.sources.split(",")
-            k = {"_cwd": args.translations_dir}
-            xgettext(
-                "--output=desktop.pot",
-                "--language=Desktop",
-                "--keyword",
-                "--keyword=Name",
-                "--package-version",
-                args.version,
-                "--msgid-bugs-address=securedrop@freedom.press",
-                "--copyright-holder=Freedom of the Press Foundation",
-                *sources,
-                **k,
+            subprocess.check_call(
+                [
+                    "xgettext",
+                    "--output=desktop.pot",
+                    "--language=Desktop",
+                    "--keyword",
+                    "--keyword=Name",
+                    "--package-version",
+                    args.version,
+                    "--msgid-bugs-address=securedrop@freedom.press",
+                    "--copyright-holder=Freedom of the Press Foundation",
+                    *sources,
+                ],
+                cwd=args.translations_dir,
             )
             msg_file_content = messages_file.read_text()
             updated_content = _remove_from_content_line_with_text(
@@ -138,7 +152,7 @@ class I18NTool:
                     if not f.endswith(".po"):
                         continue
                     po_file = os.path.join(args.translations_dir, f)
-                    msgmerge("--update", po_file, messages_file)
+                    subprocess.check_call(["msgmerge", "--update", po_file, messages_file])
                 log.warning(f"messages translations updated in {messages_file}")
             else:
                 log.warning("desktop translations are already up to date")
@@ -153,15 +167,18 @@ class I18NTool:
 
                 for source in args.sources.split(","):
                     target = source.rstrip(".in")
-                    msgfmt(
-                        "--desktop",
-                        "--template",
-                        source,
-                        "-o",
-                        target,
-                        "-d",
-                        ".",
-                        _cwd=args.translations_dir,
+                    subprocess.check_call(
+                        [
+                            "msgfmt",
+                            "--desktop",
+                            "--template",
+                            source,
+                            "-o",
+                            target,
+                            "-d",
+                            ".",
+                        ],
+                        cwd=args.translations_dir,
                     )
                     self.sort_desktop_template(join(args.translations_dir, target))
             finally:
@@ -263,12 +280,11 @@ class I18NTool:
         io.open(l10n_txt, mode="w").write(l10n_content)
         self.require_git_email_name(includes)
         if self.file_is_modified(l10n_txt):
-            k = {"_cwd": includes}
-            git.add("l10n.txt", **k)
+            subprocess.check_call(["git", "add", "l10n.txt"], cwd=includes)
             msg = "docs: update the list of supported languages"
-            git.commit("-m", msg, "l10n.txt", **k)
+            subprocess.check_call(["git", "commit", "-m", msg, "l10n.txt"], cwd=includes)
             log.warning(l10n_txt + " updated")
-            git_show_out = git.show(**k)
+            git_show_out = subprocess.check_output(["git", "show"], encoding="utf-8", cwd=includes)
             log.warning(git_show_out)
         else:
             log.warning(l10n_txt + " already up to date")
@@ -289,13 +305,15 @@ class I18NTool:
         self.ensure_i18n_remote(args)
 
         # Check out *all* remote changes to the LOCALE_DIRs.
-        git(
-            "-C",
-            args.root,
-            "checkout",
-            args.target,
-            "--",
-            *LOCALE_DIR.values(),
+        subprocess.check_call(
+            [
+                "git",
+                "checkout",
+                args.target,
+                "--",
+                *LOCALE_DIR.values(),
+            ],
+            cwd=args.root,
         )
 
         # Use the LOCALE_DIR corresponding to the "securedrop" component to
@@ -314,7 +332,7 @@ class I18NTool:
                 """
                 Add the file to the git index.
                 """
-                git("-C", args.root, "add", path)
+                subprocess.check_call(["git", "add", path], cwd=args.root)
                 paths.append(path)
 
             # Any translated locale may have changes that need to be staged from the
@@ -364,7 +382,7 @@ class I18NTool:
 
         log_command.extend([args.target, "--", path])
 
-        log_lines = subprocess.check_output(log_command).decode("utf-8").strip().split("\n")
+        log_lines = subprocess.check_output(log_command, encoding="utf-8").strip().splitlines()
         path_changes = [c.split("\x1e") for c in log_lines]
         path_changes = [
             c
@@ -380,9 +398,8 @@ class I18NTool:
         """
         Returns the list of commit hashes involving the path, most recent first.
         """
-        return git(
-            "--no-pager", "-C", root, "log", "--format=%H", path, _encoding="utf-8"
-        ).splitlines()
+        cmd = ["git", "--no-pager", "log", "--format=%H", path]
+        return subprocess.check_output(cmd, encoding="utf-8", cwd=root).splitlines()
 
     def translated_locales(self, path: str) -> Dict[str, str]:
         """Return a dictionary of all locale directories present in `path`, where the keys
@@ -396,9 +413,8 @@ class I18NTool:
         """Check if any of the given paths have had changed staged.  If so, commit them."""
         self.require_git_email_name(args.root)
         authors = set()  # type: Set[str]
-        diffs = "{}".format(
-            git("--no-pager", "-C", args.root, "diff", "--name-only", "--cached", *paths)
-        )
+        cmd = ["git", "--no-pager", "diff", "--name-only", "--cached", *paths]
+        diffs = subprocess.check_output(cmd, cwd=args.root, encoding="utf-8")
 
         # If nothing was changed, "git commit" will return nonzero as a no-op.
         if len(diffs) == 0:
@@ -413,8 +429,10 @@ class I18NTool:
             path_commits = self.get_path_commits(args.root, path)
             since_commit = None
             for path_commit in path_commits:
-                commit_message = "{}".format(
-                    git("--no-pager", "-C", args.root, "show", path_commit, _encoding="utf-8")
+                commit_message = subprocess.check_output(
+                    ["git", "--no-pager", "show", path_commit],
+                    encoding="utf-8",
+                    cwd=args.root,
                 )
                 m = self.updated_commit_re.search(commit_message)
                 if m:
@@ -425,7 +443,9 @@ class I18NTool:
 
         authors_as_str = "\n  ".join(sorted(authors))
 
-        current = git("-C", args.root, "rev-parse", args.target)
+        current = subprocess.check_output(
+            ["git", "rev-parse", args.target], cwd=args.root, encoding="utf-8"
+        )
         message = textwrap.dedent(
             """
         l10n: updated {name} ({code})
@@ -438,7 +458,8 @@ class I18NTool:
           commit: {current}
         """
         ).format(remote=args.url, name=name, authors=authors_as_str, code=code, current=current)
-        git("-C", args.root, "commit", "-m", message, *paths)
+        subprocess.check_call(["git", "commit", "-m", message, *paths], cwd=args.root)
+        log.debug(f"Committing with this message: {message}")
 
     def set_update_from_weblate_parser(self, subps: _SubParsersAction) -> None:
         parser = subps.add_parser("update-from-weblate", help=("Import translations from weblate"))

--- a/securedrop/requirements/python3/develop-requirements.txt
+++ b/securedrop/requirements/python3/develop-requirements.txt
@@ -311,9 +311,7 @@ docutils==0.14 \
 dparse==0.6.2 \
     --hash=sha256:8097076f1dd26c377f30d4745e6ec18fef42f3bf493933b842ac5bafad8c345f \
     --hash=sha256:d45255bda21f998bc7ddf2afd5e62505ba6134756ba2d42a84c56b0826614dfe
-    # via
-    #   -r requirements/python3/develop-requirements.in
-    #   safety
+    # via safety
 entrypoints==0.3 \
     --hash=sha256:589f874b313739ad35be6e0cd7efde2a4e9b6fea91edcc34e58ecbb8dbe56d19 \
     --hash=sha256:c70dd71abe5a8c85e55e12c19bd91ccfeec11a6e99044204511f9ed547d48451
@@ -977,9 +975,7 @@ ujson==5.5.0 \
     --hash=sha256:f5179088ef6487c475604b7898731a6ddeeada7702cfb2162155b016703a8475 \
     --hash=sha256:f63d1ae1ca17bb2c847e298c7bcf084a73d56d434b4c50509fb93a4b4300b0b2 \
     --hash=sha256:ff4928dc1e9704b567171c16787238201fdbf023665573c12c02146fe1e02eec
-    # via
-    #   -r requirements/python3/develop-requirements.in
-    #   python-lsp-jsonrpc
+    # via python-lsp-jsonrpc
 urllib3==1.26.6 \
     --hash=sha256:39fb8672126159acb139a7718dd10806104dec1e2f0f6c88aab05d17df10c8d4 \
     --hash=sha256:f57b4c16c62fa2760b7e3d97c35b255512fb6b59a259730f36ba32ce9f8e342f

--- a/securedrop/requirements/python3/securedrop-app-code-requirements.in
+++ b/securedrop/requirements/python3/securedrop-app-code-requirements.in
@@ -21,7 +21,6 @@ qrcode
 redis>=3.3.6
 rq>=1.8.1
 setuptools>=56.0.0
-sh
 SQLAlchemy>=1.3.0
 typing;python_version<"3.8"
 Werkzeug>=2.0.2

--- a/securedrop/requirements/python3/securedrop-app-code-requirements.txt
+++ b/securedrop/requirements/python3/securedrop-app-code-requirements.txt
@@ -276,10 +276,6 @@ rq==1.10.0 \
     --hash=sha256:92950a3e60863de48dd1800882939bbaf089a37497ebf9f2ecf7c9fd0a4c4a95 \
     --hash=sha256:be09ec43fae9a75a4d26ea3cd520e5fa3ea2ea8cf481be33e6ec9416f0369cac
     # via -r requirements/python3/securedrop-app-code-requirements.in
-sh==1.12.14 \
-    --hash=sha256:ae3258c5249493cebe73cb4e18253a41ed69262484bad36fdb3efcb8ad8870bb \
-    --hash=sha256:b52bf5833ed01c7b5c5fb73a7f71b3d98d48e9b9b8764236237bdc7ecae850fc
-    # via -r requirements/python3/securedrop-app-code-requirements.in
 six==1.11.0 \
     --hash=sha256:70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9 \
     --hash=sha256:832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb

--- a/securedrop/tests/test_i18n.py
+++ b/securedrop/tests/test_i18n.py
@@ -17,6 +17,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 import re
+import subprocess
 from pathlib import Path
 from typing import List
 
@@ -31,7 +32,6 @@ from flask import render_template, render_template_string, request, session
 from flask_babel import gettext
 from i18n import parse_locale_set
 from sdconfig import FALLBACK_LOCALE
-from sh import pybabel
 from tests.functional.factories import SecureDropConfigFactory
 from tests.functional.sd_config_v2 import DEFAULT_SECUREDROP_ROOT, SecureDropConfig
 from werkzeug.datastructures import Headers
@@ -259,7 +259,7 @@ def test_i18n():
     )
 
     pot = translation_dirs / "messages.pot"
-    pybabel("init", "-i", pot, "-d", translation_dirs, "-l", "en_US")
+    subprocess.check_call(["pybabel", "init", "-i", pot, "-d", translation_dirs, "-l", "en_US"])
 
     for (locale, translated_msg) in (
         ("fr_FR", "code bonjour"),
@@ -268,7 +268,7 @@ def test_i18n():
         ("nb_NO", "code norwegian"),
         ("es_ES", "code spanish"),
     ):
-        pybabel("init", "-i", pot, "-d", translation_dirs, "-l", locale)
+        subprocess.check_call(["pybabel", "init", "-i", pot, "-d", translation_dirs, "-l", locale])
 
         # Populate the po file with a translation
         po_file = translation_dirs / locale / "LC_MESSAGES" / "messages.po"

--- a/securedrop/tests/test_i18n_tool.py
+++ b/securedrop/tests/test_i18n_tool.py
@@ -4,6 +4,7 @@ import io
 import os
 import shutil
 import signal
+import subprocess
 import time
 from os.path import abspath, dirname, exists, getmtime, join, realpath
 from pathlib import Path
@@ -11,7 +12,6 @@ from pathlib import Path
 import i18n_tool
 import pytest
 from mock import patch
-from sh import git, msginit, pybabel, touch
 from tests.test_i18n import set_msg_translation_in_po_file
 
 
@@ -81,8 +81,17 @@ class TestI18NTool(object):
             ("ro", "SOURCE RO"),
         ]:
             po_file = Path(tmpdir) / f"{locale}.po"
-            msginit(
-                "--no-translator", "--locale", locale, "--output", po_file, "--input", messages_file
+            subprocess.check_call(
+                [
+                    "msginit",
+                    "--no-translator",
+                    "--locale",
+                    locale,
+                    "--output",
+                    po_file,
+                    "--input",
+                    messages_file,
+                ]
             )
             set_msg_translation_in_po_file(
                 po_file=po_file,
@@ -139,7 +148,18 @@ class TestI18NTool(object):
 
         locale = "en_US"
         locale_dir = join(str(tmpdir), locale)
-        pybabel("init", "-i", messages_file, "-d", str(tmpdir), "-l", locale)
+        subprocess.check_call(
+            [
+                "pybabel",
+                "init",
+                "-i",
+                messages_file,
+                "-d",
+                str(tmpdir),
+                "-l",
+                locale,
+            ]
+        )
 
         # Add a dummy translation
         po_file = Path(locale_dir) / "LC_MESSAGES/messages.po"
@@ -185,7 +205,9 @@ class TestI18NTool(object):
         locale = "en_US"
         locale_dir = join(str(tmpdir), locale)
         po_file = join(locale_dir, "LC_MESSAGES/messages.po")
-        pybabel(["init", "-i", messages_file, "-d", str(tmpdir), "-l", locale])
+        subprocess.check_call(
+            ["pybabel", "init", "-i", messages_file, "-d", str(tmpdir), "-l", locale]
+        )
         assert exists(po_file)
         # pretend this happened a few seconds ago
         few_seconds_ago = time.time() - 60
@@ -241,25 +263,25 @@ class TestI18NTool(object):
             assert b"template hello i18n" not in mo
 
     def test_require_git_email_name(self, tmpdir):
-        k = {"_cwd": str(tmpdir)}
-        git("init", **k)
+        k = {"cwd": str(tmpdir)}
+        subprocess.check_call(["git", "init"], **k)
         with pytest.raises(Exception) as excinfo:
             i18n_tool.I18NTool.require_git_email_name(str(tmpdir))
         assert "please set name" in str(excinfo.value)
 
-        git.config("user.email", "you@example.com", **k)
-        git.config("user.name", "Your Name", **k)
+        subprocess.check_call(["git", "config", "user.email", "you@example.com"], **k)
+        subprocess.check_call(["git", "config", "user.name", "Your Name"], **k)
         assert i18n_tool.I18NTool.require_git_email_name(str(tmpdir))
 
     def test_update_docs(self, tmpdir, caplog):
-        k = {"_cwd": str(tmpdir)}
-        git.init(**k)
-        git.config("user.email", "you@example.com", **k)
-        git.config("user.name", "Your Name", **k)
+        k = {"cwd": str(tmpdir)}
+        subprocess.check_call(["git", "init"], **k)
+        subprocess.check_call(["git", "config", "user.email", "you@example.com"], **k)
+        subprocess.check_call(["git", "config", "user.name", "Your Name"], **k)
         os.makedirs(join(str(tmpdir), "docs/includes"))
-        touch("docs/includes/l10n.txt", **k)
-        git.add("docs/includes/l10n.txt", **k)
-        git.commit("-m", "init", **k)
+        subprocess.check_call(["touch", "docs/includes/l10n.txt"], **k)
+        subprocess.check_call(["git", "add", "docs/includes/l10n.txt"], **k)
+        subprocess.check_call(["git", "commit", "-m", "init"], **k)
 
         i18n_tool.I18NTool().main(["--verbose", "update-docs", "--docs-repo-dir", str(tmpdir)])
         assert "l10n.txt updated" in caplog.text
@@ -271,23 +293,23 @@ class TestI18NTool(object):
         d = str(tmpdir)
         for repo in ("i18n", "securedrop"):
             os.mkdir(join(d, repo))
-            k = {"_cwd": join(d, repo)}
-            git.init(**k)
-            git.config("user.email", "you@example.com", **k)
-            git.config("user.name", "Loïc Nordhøy", **k)
-            touch("README.md", **k)
-            git.add("README.md", **k)
-            git.commit("-m", "README", "README.md", **k)
+            k = {"cwd": join(d, repo)}
+            subprocess.check_call(["git", "init"], **k)
+            subprocess.check_call(["git", "config", "user.email", "you@example.com"], **k)
+            subprocess.check_call(["git", "config", "user.name", "Loïc Nordhøy"], **k)
+            subprocess.check_call(["touch", "README.md"], **k)
+            subprocess.check_call(["git", "add", "README.md"], **k)
+            subprocess.check_call(["git", "commit", "-m", "README", "README.md"], **k)
         for o in os.listdir(join(self.dir, "i18n")):
             f = join(self.dir, "i18n", o)
             if os.path.isfile(f):
                 shutil.copyfile(f, join(d, "i18n", o))
             else:
                 shutil.copytree(f, join(d, "i18n", o))
-        k = {"_cwd": join(d, "i18n")}
-        git.add("securedrop", "install_files", **k)
-        git.commit("-m", "init", "-a", **k)
-        git.checkout("-b", "i18n", "master", **k)
+        k = {"cwd": join(d, "i18n")}
+        subprocess.check_call(["git", "add", "securedrop", "install_files"], **k)
+        subprocess.check_call(["git", "commit", "-m", "init", "-a"], **k)
+        subprocess.check_call(["git", "checkout", "-b", "i18n", "master"], **k)
 
         def r():
             return "".join([str(l) for l in caplog.records])
@@ -350,7 +372,11 @@ class TestI18NTool(object):
         )
         assert "l10n: updated Dutch (nl)" not in r()
         assert "l10n: updated German (de_DE)" not in r()
-        message = str(git("--no-pager", "-C", "securedrop", "show", _cwd=d, _encoding="utf-8"))
+        message = subprocess.check_output(
+            ["git", "--no-pager", "-C", "securedrop", "show"],
+            cwd=d,
+            encoding="utf-8",
+        )
         assert "Loïc" in message
 
         # an update is done to nl in weblate
@@ -362,14 +388,15 @@ class TestI18NTool(object):
         updated_content = content.replace(text_to_update, "INACTIVITEIT")
         po_file.write_text(updated_content)
 
-        git.add(str(po_file), **k)
-        git.config("user.email", "somone@else.com", **k)
-        git.config("user.name", "Someone Else", **k)
-        git.commit("-m", "translation change", str(po_file), **k)
+        k = {"cwd": join(d, "i18n")}
+        subprocess.check_call(["git", "add", str(po_file)], **k)
+        subprocess.check_call(["git", "config", "user.email", "somone@else.com"], **k)
+        subprocess.check_call(["git", "config", "user.name", "Someone Else"], **k)
+        subprocess.check_call(["git", "commit", "-m", "translation change", str(po_file)], **k)
 
-        k = {"_cwd": join(d, "securedrop")}
-        git.config("user.email", "somone@else.com", **k)
-        git.config("user.name", "Someone Else", **k)
+        k = {"cwd": join(d, "securedrop")}
+        subprocess.check_call(["git", "config", "user.email", "somone@else.com"], **k)
+        subprocess.check_call(["git", "config", "user.name", "Someone Else"], **k)
 
         #
         # the nl translation update from weblate is copied
@@ -390,6 +417,10 @@ class TestI18NTool(object):
         )
         assert "l10n: updated Dutch (nl)" in r()
         assert "l10n: updated German (de_DE)" not in r()
-        message = str(git("--no-pager", "-C", "securedrop", "show", _cwd=d))
+        message = subprocess.check_output(
+            ["git", "--no-pager", "-C", "securedrop", "show"],
+            cwd=d,
+            encoding="utf-8",
+        )
         assert "Someone Else" in message
         assert "Loïc" not in message

--- a/securedrop/tests/test_template_filters.py
+++ b/securedrop/tests/test_template_filters.py
@@ -1,3 +1,4 @@
+import subprocess
 from datetime import datetime, timedelta
 from pathlib import Path
 
@@ -8,7 +9,6 @@ import source_app
 import template_filters
 from db import db
 from flask import session
-from sh import pybabel
 from tests.test_i18n import create_config_for_i18n_test
 
 
@@ -104,7 +104,7 @@ def do_test(create_app):
 
     for l in ("en_US", "fr_FR"):
         pot = Path(test_config.TEMP_DIR) / "messages.pot"
-        pybabel("init", "-i", pot, "-d", test_config.TEMP_DIR, "-l", l)
+        subprocess.check_call(["pybabel", "init", "-i", pot, "-d", test_config.TEMP_DIR, "-l", l])
 
     app = create_app(test_config)
     with app.app_context():

